link/src/main/java/com/stripe/android/link/ui/wallet/WalletUiState.kt

```diff
@@ -1,24 +1,38 @@
 package com.stripe.android.link.ui.wallet
 
 import androidx.compose.runtime.Immutable
+import com.stripe.android.core.strings.ResolvableString
 import com.stripe.android.link.ui.PrimaryButtonState
 import com.stripe.android.model.ConsumerPaymentDetails
+import com.stripe.android.model.ConsumerPaymentDetails.Card
 
 @Immutable
 internal data class WalletUiState(
     val paymentDetailsList: List<ConsumerPaymentDetails.PaymentDetails>,
     val selectedItem: ConsumerPaymentDetails.PaymentDetails?,
     val isProcessing: Boolean,
+    val primaryButtonLabel: ResolvableString,
+    val hasCompleted: Boolean,
 ) {
 
     val showBankAccountTerms = selectedItem is ConsumerPaymentDetails.BankAccount
 
     val primaryButtonState: PrimaryButtonState
         get() {
-            return if (isProcessing) {
+            val card = selectedItem as? Card
+            val isExpired = card?.isExpired ?: false
+            val requiresCvcRecollection = card?.cvcCheck?.requiresRecollection ?: false
+
+            val disableButton = isExpired || requiresCvcRecollection
```

Comment:
how will a user enter their CVC if the pay button is disabled when CVC is required?

Reply:
There will be a form asking the user for this when they attempt to pay. This condition will also be updated.

[JIRA](https://jira.corp.stripe.com/browse/MOBILESDK-2836)